test/setup-beam.test.js

```diff
@@ -171,6 +174,9 @@ async function testOTPVersions() {
   )
 
   if (process.platform === 'linux') {
+    const previousRunnerArch = process.env.RUNNER_ARCH
+    process.env.RUNNER_ARCH = 'X64'
```

Comment:
Not sure what we're doing here.

I'd propose you duplicate `check_integrity` (from the workflows) to `check_integrity-amd` and `check_integrity-arm` and then the code itself (even in the tests) decides what is ARM and what is AMD (as per the functions you added below: `testLinuxARM64OTPVersions`, `testLinuxAMD64OTPVersions`) otherwise it seems we're forcing the ARM tests to run in AMD, and we can probably do better 😄

Reply:
Hm... it seems ARM runners are not yet generally available.